featurebyte/api/target.py

```diff
@@ -0,0 +1,82 @@
+"""
+Target API object
+"""
+from __future__ import annotations
+
+from typing import Any, List, Optional
+
+from pydantic import Field
+
+from featurebyte.api.api_object import ForeignKeyMapping
+from featurebyte.api.entity import Entity
+from featurebyte.api.savable_api_object import SavableApiObject
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.models.target import TargetModel
+from featurebyte.schema.target import TargetUpdate
+
+
+class Target(SavableApiObject):
+    """
+    Target class used to represent a Target in FeatureByte.
+    """
+
+    internal_entity_ids: Optional[List[PydanticObjectId]] = Field(alias="entity_ids")
+    horizon: Optional[str]
+    blind_spot: Optional[str]
+
+    _route = "/target"
+    _update_schema_class = TargetUpdate
+
+    _list_schema = TargetModel
+    _get_schema = TargetModel
+    _list_fields = ["name", "entities"]
+    _list_foreign_keys = [
+        ForeignKeyMapping("entity_ids", Entity, "entities"),
+    ]
+
+    @property
+    def entities(self) -> List[Entity]:
+        """
+        Returns a list of entities associated with this target.
+
+        Returns
+        -------
+        List[Entity]
+        """
+        if not self.internal_entity_ids:
+            return []
+        return [Entity.get_by_id(entity_id) for entity_id in self.internal_entity_ids]
```

Comment:
We should use the pattern similar to https://github.com/featurebyte/featurebyte/blob/6e4db13c617a4b234f5efde2b6967137ee3335f1/featurebyte/api/event_table.py#L217-L229

The main idea is that we use `entity_ids` from the persistent object if it exists. Otherwise, use the one stored at the memory.